src/ucp/core/ucp_worker.c

```diff
@@ -1497,30 +1496,25 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
         return UCS_ERR_NO_MEMORY;
     }
 
+    uct_thread_mode = UCS_THREAD_MODE_SINGLE;
+    worker->flags   = 0;
+
     if (params->field_mask & UCP_WORKER_PARAM_FIELD_THREAD_MODE) {
 #if !ENABLE_MT
-        thread_mode = UCS_THREAD_MODE_SINGLE;
+        uct_thread_mode = UCS_THREAD_MODE_SINGLE;
         if (params->thread_mode != UCS_THREAD_MODE_SINGLE) {
             ucs_debug("forced single thread mode on worker create");
         }
 #else
-        thread_mode = params->thread_mode;
-#endif
-    } else {
-        thread_mode = UCS_THREAD_MODE_SINGLE;
-    }
-
-    if (thread_mode == UCS_THREAD_MODE_MULTI) {
-        worker->flags = UCP_WORKER_FLAG_MT;
-    } else {
-        worker->flags = 0;
-    }
+        if (params->thread_mode != UCS_THREAD_MODE_SINGLE) {
+            /* UCT is serialized by UCP lock or by UCP user */
+            uct_thread_mode = UCS_THREAD_MODE_SERIALIZED;
+        }
 
-    if (thread_mode == UCS_THREAD_MODE_SINGLE) {
-        uct_thread_mode = UCS_THREAD_MODE_SINGLE;
-    } else {
-        /* UCT is serialized by UCP lock or by UCP user */
-        uct_thread_mode = UCS_THREAD_MODE_SERIALIZED;
+        if (params->thread_mode == UCS_THREAD_MODE_MULTI) {
+            worker->flags = UCP_WORKER_FLAG_MT;
+        }
+#endif
```

Comment:
can we swap the if/else of ENABLE_MT
e.g `#if ENABLE_MT` instead of `#if !ENABLE_MT`

Reply:
done